### PR TITLE
[qontract-cli] get maintenances

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-10234

this qontract-cli command will list maintenances

### CODE REVIEW

File: tools/qontract_cli.py [hunk 1]
```diff
@@ -2577,6 +2578,27 @@ def osd_component_versions(ctx):
...
+@click.pass_context
+def maintenances(ctx):
+    now = datetime.now(timezone.utc)
+    maintenances = maintenances_gql.query(gql.get_api().query).maintenances or []
+    maintenances = [
+        m for m in maintenances if datetime.fromisoformat(m.scheduled_end) > now
+    ]
+    data = [vars(m) for m in maintenances]
+    for m in data:
+        m["services"] = ", ".join([a.name for a in m["affected_services"]])
```
Comment (hemslo) [inline tools/qontract_cli.py:2591]: the generated gql basemodel has a method `dict` to convert model to dict, also support cherry pick fields if needed, [doc](https://docs.pydantic.dev/1.10/usage/exporting_models/#modeldict).

so we can do this data conversion in one go:

```python
    data = [
        {
            **m.dict(),
            "services": ", ".join(a.name for a in m.affected_services),
        }
        for m in maintenances
    ]
```

File: tools/qontract_cli.py [hunk 2]
```diff
@@ -2577,6 +2578,27 @@ def osd_component_versions(ctx):
...
+    print(data)
```
Comment (hemslo) [inline tools/qontract_cli.py:2592]: Is this for debug print or we want to pretty print `data`?

File: tools/qontract_cli.py (hunk 1 above)
Comment (maorfr) [reply tools/qontract_cli.py]: https://github.com/app-sre/qontract-reconcile/pull/4348/commits/4178ba8c79eec8cc0e569b7d1d9b53cc73b91a04

File: tools/qontract_cli.py (hunk 2 above)
Comment (maorfr) [reply tools/qontract_cli.py]: debugging leftovers. https://github.com/app-sre/qontract-reconcile/pull/4348/commits/4178ba8c79eec8cc0e569b7d1d9b53cc73b91a04

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**